zap/src/main/java/org/parosproxy/paros/core/scanner/AbstractPlugin.java

```diff
@@ -277,15 +277,11 @@ protected void sendAndReceive(
                                 .getExtension(ExtensionAntiCSRF.class);
             }
             if (extAntiCSRF != null) {
-                List<AntiCsrfToken> tokens = extAntiCSRF.getTokens(message);
-                AntiCsrfToken antiCsrfToken = null;
-                if (tokens.size() > 0) {
-                    antiCsrfToken = tokens.get(0);
-                }
-
-                if (antiCsrfToken != null) {
-                    regenerateAntiCsrfToken(message, antiCsrfToken);
+                if (!isRegenCsrfCalled) {
```

Comment:
It can be sent with the correct parameters instead, e.g.:
```Java
extAntiCSRF.regenerateAntiCsrfToken(message, tokenMsg -> sendAndReceive(tokenMsg, true, false));
```